apps/api/src/app/workflows-v2/usecases/get-workflow/get-workflow.usecase.ts

```diff
@@ -21,6 +32,29 @@ export class GetWorkflowUseCase {
       })
     );
 
-    return toResponseWorkflowDto(workflowEntity);
+    const fullSteps = await this.getFullWorkflowSteps(workflowEntity, command.user);
+
+    return toResponseWorkflowDto(workflowEntity, fullSteps);
+  }
+
+  private async getFullWorkflowSteps(
+    workflow: WorkflowInternalResponseDto,
+    user: UserSessionData
```

Comment:
Regarding my first comment, maybe it's time to kill this mapper, and have it map inside the get use case and just return it as the response of upsert

Reply:
Yes could be an option, but I also think its fine for now if internally we use the `WorkflowInternalResponseDto` which is returned from the upsert etc, and use the full step data just for external DTO here. 

We can re-evaluate later, but for now I think this is the simplest and safest option.